### ISSUE TITLE
Bug: [prefer-nullish-coalescing] bad handling of computed members

### ISSUE THREAD

**Issue by OlivierZal** [typescript-eslint/typescript-eslint#10890]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.7.2&fileType=.tsx&code=MYewdgzgLgBAhjAvDA5AIxQWAFA4CYCmwANnAE4EzEGwAeAXDAN7yPRkCWYA5gDQxo2UTjxgBfGAB8YYAK7FiOHLQD8AOgQBCZHIUwVMWgG04AXRiMUAMxAgsubAVoAHEGVgsxQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6KL2jjokcGAC%2BILUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
const a = 'b';

declare let x: { a: string, b: string } | null

x?.a != null ? x[a] : 'foo'
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/prefer-nullish-coalescing": ["error"],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

`x?.a != null ? x[a] : 'foo'` should not report

### Actual Result

`x?.a != null ? x[a] : 'foo'` reports

### Additional Info

Raised in https://github.com/typescript-eslint/typescript-eslint/issues/10733#issuecomment-2680821235

Root cause: the rule doesn't handle well computed members. 

A fix is ready in the perimeter of https://github.com/typescript-eslint/typescript-eslint/pull/10861: https://github.com/typescript-eslint/typescript-eslint/commit/f69b588a1451a1d2cac58ba45248d06c4d2bf855

✅ [PR branch playground](https://deploy-preview-10861--typescript-eslint.netlify.app/play/#ts=5.7.2&fileType=.tsx&code=MYewdgzgLgBAhjAvDA5AIxQWAFA4CYCmwANnAE4EzEGwAeAXDAN7yPRkCWYA5gDQxo2UTjxgBfGAB8YYAK7FiOHLQD8AOgQBCZHIUwVMWgG04AXRiMUAMxAgsubAVoAHEGVgsxQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Y6RAM0WlqYSNkAC1pkA9gEMkyMswDm6KL2jjokcGAC%2BILUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

Cc @kirkwaiblinger

**Comment by OlivierZal** [comment]
@JoshuaKGoldberg, @kirkwaiblinger, this one can be closed given that https://github.com/typescript-eslint/typescript-eslint/pull/10861 has been merged.
